Bluetooth: Mesh: Adds three Config Client API

haoyanbiao opened the pull request:
for deleting appkey, unbinding an application from SIG model, unbinding an application from vendor model on the target node, withmatching shell command.
Signed-off-by: haoyanbiao <haoyanbiao@xiaomi.com>

trond-snekvik commented on include/bluetooth/mesh/cfg_cli.h:
```diff
@@ -273,6 +273,20 @@ int bt_mesh_cfg_app_key_add(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_app_key_del(u16_t net_idx, u16_t addr, u16_t key_net_idx,
+			    u16_t key_app_idx, u8_t *status);
```
The Zephyr integer types have been deprecated, could you please convert these to stdint's `uint16_t` and `uint8_t`? Same throughout

trond-snekvik commented on subsys/bluetooth/mesh/cfg_cli.c:
```diff
@@ -1161,6 +1161,47 @@ int bt_mesh_cfg_app_key_get(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
+
+	bt_mesh_model_msg_init(&msg, OP_APP_KEY_DEL);
+	key_idx_pack(&msg, key_net_idx, key_app_idx);
+	net_buf_simple_add_mem(&msg, app_key, 16);
```
The key value field shouldn't be present in app_key_del.

trond-snekvik commented on subsys/bluetooth/mesh/shell.c:
```diff
@@ -1206,6 +1237,48 @@ static int cmd_mod_app_bind(const struct shell *shell, size_t argc,
+
+	return 0;
+}
+
```
Please add these two to the command list at the bottom, and add corresponding documentation entries in doc/reference/bluetooth/mesh/shell.rst :)

trond-snekvik reviewed (changes requested):
Thanks for contributing!

Looks good overall, but I have a couple comments that have to be resolved.

trond-snekvik commented:
The Compliance check also has some issues with the format of your commit message.

The title needs to follow the <module>: <subject> format, ie "Bluetooth: Mesh: Add three Config Client APIs"

Additionally, the email in the signoff-line needs to match the email you committed with. Git can insert this automatically for you if you commit with the `-s` flag.

haoyanbiao committed:
Bluetooth: Mesh: Adds three Config Client API.

Those APIs are used for deleting appkey, unbinding an application
from SIG model, unbinding an application from vendor model on the
target node, with matching shell command.

Signed-off-by: YanBiao Hao <haoyanbiao@126.com>

trond-snekvik commented on subsys/bluetooth/mesh/shell.c:
```diff
@@ -2603,6 +2676,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(mesh_cmds,
 	SHELL_CMD_ARG(app-key-add, NULL, "<NetKeyIndex> <AppKeyIndex> [val]",
 		      cmd_app_key_add, 3, 1),
+	SHELL_CMD_ARG(app-key-del, NULL, "<NetKeyIndex> <AppKeyIndex>",
+		      cmd_app_key_del, 2, 0),
```
Have to include the `mesh` parameter in the count here, so the argument count parameter should be `3` here. The `mod-app-unbind` entry is correct.

trond-snekvik reviewed (changes requested):
Thanks. I just have one additional issue, and the compliance check has a couple of minor warnings that need to be resolved.

LingaoM commented on doc/reference/bluetooth/mesh/shell.rst:
```diff
@@ -413,6 +413,15 @@ The Configuration Client uses the general messages parameters set by ``mesh dst`
+	Delete an application key from the target node.
+
+	* ``NetKeyIndex``: The network key index the application key is bound to.
+	* ``AppKeyIndex``: The application key index to add.
```
The application key index to delete.

LingaoM commented on include/bluetooth/mesh/cfg_cli.h:
```diff
@@ -287,6 +301,21 @@ int bt_mesh_cfg_app_key_get(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
+ *  @param net_idx     Network index to encrypt with.
+ *  @param addr        Target node address.
+ *  @param elem_addr   Element address the model is in.
+ *  @param mod_app_idx Application index to bind.
```
Application key index to unbind.

LingaoM replied:
Application key index to unbind.

LingaoM commented on include/bluetooth/mesh/cfg_cli.h:
```diff
@@ -303,6 +332,22 @@ int bt_mesh_cfg_mod_app_bind_vnd(uint16_t net_idx, uint16_t addr, uint16_t elem_
+ *  @param net_idx     Network index to encrypt with.
+ *  @param addr        Target node address.
+ *  @param elem_addr   Element address the model is in.
+ *  @param mod_app_idx Application index to bind.
```
Same too.

LingaoM replied:
Same too.
